### PR TITLE
fix greedy_algorithms/knapsack.cpp to satisfy clang-tidy warnings

### PR DESCRIPTION
**Pull request by xgupta**
#### Description of Change

Fx greedy_algorithms/knapsack.cpp to satisfy clang-tidy warnings.

#### Checklist

- [x] Added description of change
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1261"></a>

### CODE REVIEW

File: greedy_algorithms/knapsack.cpp [hunk 1]
```diff
@@ -58,21 +61,21 @@ int main() {
...
             break;
         }
     }
 
-    cout << "\nMax Profit : " << maxProfit;
+    std::cout << "\nMax Profit : " << maxProfit;
 
     return 0;
 }
+
```
Comment (Panquesito7) [inline greedy_algorithms/knapsack.cpp:81]:
```suggestion
```

**Review by Panquesito7 — APPROVED** [review]

File: greedy_algorithms/knapsack.cpp [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
     int profit;
 };
 
-float profitPerUnit(Item x) { return (float)x.profit / (float)x.weight; }
+float profitPerUnit(Item x) {
+    return static_cast<float>(x.profit) / static_cast<float>(x.weight);
+}
 
-int partition(Item arr[], int low, int high) {
+int partition(std::vector<Item> arr, int low, int high) {
```
Comment (kvedala) [inline greedy_algorithms/knapsack.cpp:14]:
```suggestion
int partition(const std::vector<Item> &arr, int low, int high) {
```

File: greedy_algorithms/knapsack.cpp [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
-using namespace std;
+#include <vector>
 
 struct Item {
     int weight;
     int profit;
 };
 
-float profitPerUnit(Item x) { return (float)x.profit / (float)x.weight; }
+float profitPerUnit(Item x) {
```
Comment (kvedala) [inline greedy_algorithms/knapsack.cpp:10]:
```suggestion
float profitPerUnit(const Item &x) {
```

File: greedy_algorithms/knapsack.cpp [hunk 4]
```diff
@@ -1,14 +1,17 @@
+#include <cmath>
 #include <iostream>
-using namespace std;
+#include <vector>
 
 struct Item {
```
Comment (kvedala) [inline greedy_algorithms/knapsack.cpp:5]:
```suggestion

namespace greedy_algorithms {

namespace knapsack {

struct Item {
```

File: greedy_algorithms/knapsack.cpp [hunk 5]
```diff
@@ -38,17 +41,17 @@ void quickSort(Item arr[], int low, int high) {
 }
```
Comment (kvedala) [inline greedy_algorithms/knapsack.cpp:42]:
```suggestion
} // namespace knapsack
} // namespace greedy_algorithms

using greedy_algorithms::knapsack::Item;
using greedy_algorithms::knapsack::quickSort;
```

**Review by kvedala — CHANGES_REQUESTED** [review]
would be great if you can add the documentation as well.
Thank you

**Comment by xgupta** [comment]
/home/user/C-Plus-Plus/greedy_algorithms/knapsack.cpp:28:20: error: no viable overloaded '=' [clang-diagnostic-error]
            arr[i] = arr[j];

When we make array as constant this message is produced by clang-tidy. Because we are then want to modify its value.

File: greedy_algorithms/knapsack.cpp [hunk 6]
```diff
@@ -1,14 +1,55 @@
...
+ * @file
+ * @brief [Eight Queens](https://en.wikipedia.org/wiki/Knapsack_problem)
+ *
+ * @details
+ * Given a set of items, each with a weight and a value, determine the number of
+ * each item to include in a collection so that the total weight is less than or
+ * equal to a given limit and the total value is as large as possible.
+ *
+ * @author Unknown author
+ * @author [Krishna Vedala](https://github.com/kvedala)
```
Comment (kvedala) [inline greedy_algorithms/knapsack.cpp:11]:
```suggestion
```
Thank you. I only reviewed but did not make the changes. Keep your name.

**Review by kvedala — CHANGES_REQUESTED** [review]
some suggestions have been blindly marked "resolved" while the concerns still remain in the code.
Please re-review the above suggestion.

**Pull request closed without merging**